### ISSUE TITLE
Padding and Masking Settings during Training

### ISSUE THREAD

**Issue by Ching-Yee-Chan** [kyutai-labs/moshi#126]
### Due diligence

- [X] I have done my due diligence in trying to find the answer myself.

### Topic

The paper

### Question

My question is about how maximum sequence length is set and how masked loss is computed during batched training of Moshi. Specifically, I suppose that the batched input and target of an SFT training step be in the below format:

![32e1132af19f9adde71da0b9784414e](https://github.com/user-attachments/assets/21a87960-a4b3-4096-83a1-fa6a5a49850b)

My questions are as follows:
1. Are losses at position ① summed into the final loss? If so, what are the ground truth labels of these positions? (We guess that there may be three types of padding tokens in the text layer, and the choice of the last token target may affect how Moshi determine the ending point of its response?)
2. Are losses at position ② summed into the final loss? (i.e, did you apply a padding mask during batched training and how?)
3. Are losses at position ③ summed into the final loss?
3. How did you set the maximum length T of a batch sequence? (Say, if running on an 80GB GPU)?

**Comment by dkounadis** [comment]
### Offset of Positional Embeds 

About [offsets added to RoPe](https://github.com/kyutai-labs/moshi/blob/8a49848a8b8f40173789754acd4b3f3f1bec9246/moshi/moshi/modules/transformer.py#L399). LLM pos embed has offset = 0,1,2,3,4, .. , num_txt_tokens. Mimi Encoder pos embed has offset = 0,40,80,120,160, .. , 40xnum_enco_toks. Mimi Decoder pos embed has offset = 0,1,2,3,4, .. , num_deco_tokens. Depformer pos embed has offset = 0, .., 7 

Please tell me if I miss an increment!
